content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md

```diff
@@ -15,151 +15,215 @@ keywords:
+
+* Create a [wallet address](#getting-a-wallet-address).
 
-2. Update [TrueNAS SCALE]({{< relref "UpdateSCALE.md" >}}) to the latest public release.
+* Obtain a [Storj authentication token](#generating-an-authentication-token-for-storj).
 
-3. Create a [wallet address](#getting-a-wallet-address).
+* [Configure your router and firewall](#configuring-the-router-and-firewall).
 
-4. Generate a [Storj authentication token](#generating-an-authentication-token-for-storj).
+* [Create a publicly-available domain name](#creating-a-ddns-host-name) to access the Storj application. Point this to your router public IP address.
 
-5. Configure your router and firewall.
-   Open ports on your router and configure port forwarding. Configure firewall rules to [allow access for these ports](#configuring-the-router-and-firewall).
-   * Default 20988
-   * Open 28967 for both TCP and UDP.
-   * Open ports 7777 and 8888 for outbound communication.
+* [Create a Storj identity and authorize it](https://docs.storj.io/node/dependencies/identity) for every node.
 
-   Alternatively, use a dynamic DNS (DDNS) service such as NoIP to [create a host name](#creating-a-ddns-host-name) if you do not have a static IP address for the system nodes.
+  Every node must have a unique identifier on the network. Use NFS/SMB shares or a file transfer service such as FTP to upload the credentials generated.
+  If the identity is not present in the storage directory, it generates and authorizes one automatically.
+  This can take a long time and consume system resources while it generates one.
 
-6. Create a publicly-available domain name to access the Storj application. Point this to your router public IP address.
+* Prepare TrueNAS SCALE for the app installation by [creating the datasets](#creating-the-storj-datasets-on-truenas-scale) for the application storage, and a new user if not using the default apps user.
+   
+   Create a parent dataset, such as *storj-node* and then the two storage datasets (**identity** and **config**) under it.
+   Select **apps** as the **Dataset Preset** for these datasets. You can modify the dataset ACLs at the time of creation, or modify them later when adding them in the app.
 
-7. [Create a Storj identity and authorize it](https://docs.storj.io/node/dependencies/identity) for every node.
-   Every node must have a unique identifier on the network. Use NFS/SMB shares or a file transfer service such as FTP to upload the credentials generated.
-   If the identity is not present in the storage directory, it generates and authorizes one automatically.
-   This can take a long time and consume system resources while it generates one.
+   If not using the default user **apps** (568) as the assigned user and group, go to **Credentials > Local User**, click **Add** and create the new user.
+   Make sure the user has read/write/execute privileges. If you have an existing admin user on your system with the right permissions, you make that user the app user.
```

Comment:
execute shouldn't be needed for **data **

Reply:
I'll add the note about the **data** dataset permissions in the section on the ACL. This part is talking about the user permissions in general.

Reply:
```suggestion
   Make sure the user has read/write/execute privileges. If you have an existing admin user on your system with the right permissions, you can make that user the app user.
```